python/lsst/ip/isr/linearize.py

```diff
@@ -51,43 +48,65 @@ class Linearizer(abc.ABC):
             - one column for each image value
         To avoid copying the table the last index should vary fastest
         (numpy default "C" order)
-    detector : `lsst.afw.cameraGeom.Detector`
-        Detector object
-    override : `bool`, optional
-        Override the parameters defined in the detector/amplifier.
+    detector : `lsst.afw.cameraGeom.Detector`, optional
+        Detector object.  Passed to self.fromDetector() on init.
     log : `lsst.log.Log`, optional
         Logger to handle messages.
+    kwargs : `dict`, optional
+        Other keyword arguments to pass to the parent init.
 
     Raises
     ------
     RuntimeError :
         Raised if the supplied table is not 2D, or if the table has fewer
         columns than rows (indicating that the indices are swapped).
-    """
 
-    _OBSTYPE = "linearizer"
-    """The dataset type name used for this class"""
+    Notes
+    -----
+    The linearizer attributes stored are:
+
+    hasLinearity : `bool`
+        Whether a linearity correction is defined for this detector.
+    override : `bool`
+        Whether the detector parameters should be overridden.
+    ampNames : `list` [`str`]
+        List of amplifier names to correct.
+    linearityCoeffs : `dict` [`str`, `numpy.array`]
+        Coefficients to use in correction.  Indexed by amplifier
+        names.  The format of the array depends on the type of
+        correction to apply.
+    linearityType : `dict` [`str`, `str`]
+        Type of correction to use, indexed by amplifier names.
+    linearityBBox : `dict` [`str`, `lsst.geom.Box2I`]
```

Comment:
What is the exact meaning of this bounding box? what are the regions where the correction is valid and where it is not? I'm a bit confused by the explanation n the docstring (sorry!)

Reply:
Again, carryover from the `cameraGeom` definitions.  The individual amplifiers have been assembled and trimmed so this is used to define the region to apply that amplifier's correction.  This means that the correction can be run without passing a detector (which also defines the BBox).  The alternative is to make `linearizer.applyLinearity()` require the `detector` argument.